core/types/calls.cc

```diff
@@ -1461,36 +1461,54 @@ class T_attached_class : public IntrinsicMethod {
     }
 } T_attached_class;
 
-class T_must : public IntrinsicMethod {
-public:
-    void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
-        if (args.args.empty()) {
-            return;
+namespace {
+
+enum class MustBehavior {
+    T,
+    Magic,
```

Comment:
Slight preference for naming this according to what functionality it gates (pedantic redundancy check vs not) instead of tracking which Ruby method uses which (not saying this will necessarily happen soon, but imagine adding one more intrinsic that uses `implementMust`: it'd probably want to specify which behavior to get, like "give me pedantic checks," rather than have to say "give me the `T.must` behavior."